addons/pos_loyalty/models/loyalty_card.py

```diff
@@ -19,7 +19,7 @@ def _get_default_template(self):
         return super()._get_default_template()
 
     def _get_mail_partner(self):
-        return super()._get_mail_partner() or self.source_pos_order_id.partner_id
+        return super()._get_mail_partner() or self.sudo().source_pos_order_id.partner_id
```

Comment:
A gift card created in one company with explicit company in it, shouldn't be used in different company. Since the program linked to it doesn't have a company, what if we create the gift card without specifying the company instead of using sudo here?

Reply:
@caburj I’ve explained this before, but I apologize if it wasn’t clear. The issue is that the gift card doesn’t have a company associated with it, whereas the PoS order does. This makes accessing source_pos_order_id problematic.